src/SSR.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from "react";
+
+const PassThrought: any = ({children}: { children: React.ReactNode }) => children;
+
+export class MediaServerSide extends React.Component<{
+  predicted: string,
+  fact: string,
+  hydrated: boolean;
+  children: React.ReactNode
+}, {
+  key: string
+}> {
+  state = {
+    key: 'media-as-predicted'
+  };
+
+  componentDidMount() {
+    const {fact, predicted, hydrated } = this.props;
+    if (fact && fact !== predicted) {
+      if(hydrated) {
```

Comment:
If predicted !== fact - throw an error and change to level key, ie remount everything